### ISSUE TITLE
Pregunta flujo Tarea 2

### ISSUE THREAD

**Issue by SebaHagedorn** [DiegoEmilio01/IIC3413#15]
Hola! Tengo una pregunta sobre el flujo de tarea 2. Está como primer problema en el enunciado implementar la búsqueda, pero entiendo que es necesario implementar el insert_record del IsamNonClusteredLeaf antes para que se pueda buscar algo, estoy en lo correcto? Por la misma línea,  al insertar los records en el test, pensé que el flujo pasaba por la función mencionada pero no funciona mi print dentro del insert.

Todo esto surgió porque al buscar en el Iter, me sale el valor de N igual a 0 y no me hace sentido.

Muchas gracias de antemano.

**Comment by cirojas** [comment]
Sí, ambas operaciones dependen de la otra. Para probar si funciona el iter vas a tener que implementar la inserción (o crear hojas de prueba a manualmente). Y para probar si funciona bien la inserción tendrás que implementar la búsqueda (o revisar directamente los bytes de la página).

**Comment by SebaHagedorn** [comment]
Gracias!

**Comment by SebaHagedorn** [comment]
La única duda que me queda es que cuando ejecuto el test2, no pasa por este método (porque intento printear):

`void IsamNonClusteredLeaf::insert_record(RID rid, int64_t key)`

entonces me queda la duda si tengo que modificar alguna parte del flujo a parte de lo señalado como el TODO para que guarde los records en la hoja.

**Comment by cirojas** [comment]
Puede ser que te quedó mal el archivo desde una creación anterior. El método 

`void IsamNonClusteredLeaf::insert_record(RID rid, int64_t key)`

deberia llamarse en el constructor de `IsamNonClustered`, pero en el constructor si el archivo ya existe no se crea.
Borra la carpeta de la base de datos e intenta de nuevo.

**Comment by SebaHagedorn** [comment]
Ahora sí, está pasando. Gracias por la ayuda.
